client/src/i18n.js

```diff
@@ -90,6 +93,11 @@ class i18n {
     for (let entity in dict) {
       this.lang[locale][entity] = dict[entity];
     }
+
+    // Re-set current locale in case the new dictionary provides a better match than the old locale.
+    if (this.autoDetectLocale) {
+      this.currentLocale = this.detectLocale();
+    }
```

Comment:
Dictionaries are added by separate `.js` files. This is the most reliable way to ensure that the current locale is detected according to all dictionaries that will be loaded in, allowing for slow network speeds. i.e. at the time each dictionary is loaded in, the most-correct current locale will be set.

We have no way of knowing ahead of time if more dictionaries will be loaded, so we can't reasonably defer until all of them are loaded.